### PR TITLE
Be2021 google doc as yaml

### LINKED ISSUE #437: BE2021 google doc as yaml

### PR DESCRIPTION
**Pull request by baob**
Closes #437 .

New yaml file created containing data from the BE2021 google doc (candidates, polls, constituencies, parties)

db/seed.rb replaced for the by election. This can be run on a development environment with 

     rake db:drop && rake db:create && rake db:schema:load && rake db:seed
     
which will completely destroy your existing dev database of course, so take a backup first if need.

I've verified correctness _only_ by inspecting the database.   I have not as yet investigated the impact on running code.

### CODE REVIEW

**Comment by baob** [comment]
Found a problem: On choosing a postcode not in one of the two constituencies, this happens

image

**Comment by aspiers** [comment]
[**@baob**](https://github.com/baob) commented on [June 5, 2021 7:44 PM](https://github.com/swapmyvote/swapmyvote/pull/439#issuecomment-855279897 "2021-06-05T18:44:29Z - Replied by Github Reply Comments"):
> Found a problem: On choosing a postcode not in one of the two constituencies, this happens
> 
> image

I suggest filing a separate issue for that, labelled `BE2021`.

**Comment by baob** [comment]
> I suggest filing a separate issue for that, labelled `BE2021`.

Done

**Review by aspiers — APPROVED** [review]
Awesome work! Especial kudos for all the tests :sunglasses:

File: db/fixtures/be2021/party.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            @instance ||= new
+          end
+
+          def all
+            instance.all
+          end
+        end
+
+        def all_with_duplicates
+          Candidate.new.all.map do |candidate|
```
[the remark this answers is not in the document]
Comment (baob) [reply db/fixtures/be2021/party.rb:19]: Agreed ... I started with a singleton pattern I was familiar with and only at the end realised all the extra weight it added for this case. Raised PR anyway due to the urgency.

File: db/fixtures/be2021_yaml.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+# This is an encapsulation of the Yaml file for By-election 2019
+
+module Db
+  module Fixtures
+    class Be2021Yaml
+      FILENAME = "db/fixtures/be2021.yml"
+
+      class << self
+        def instance
```
[the remark this answers is not in the document]
Comment (baob) [reply db/fixtures/be2021_yaml.rb:11]: Agreed ... I started with a singleton pattern I was familiar with and only at the end realised all the extra weight it added for this case. Raised PR anyway due to the urgency.

File: db/fixtures/be2021/party.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (aspiers) [reply db/fixtures/be2021/party.rb:19]: You did the right thing :-)

**Pull request merged**